### PR TITLE
first header comments

### LINKED ISSUE #93: documentation for cbl_LAI_Canopy

### PR DESCRIPTION
**Pull request by har917**
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

Documentation for the routine cbl_LAI_canopy_height.F90

Fixes #93 

## Type of change

- [ ] Bug fix
- [x] New or updated documentation

## Checklist

- [ ] The new content is accessible and located in the appropriate section.
- [x] I have checked that links are valid and point to the intended content.
- [x] I have checked my code/text and corrected any misspellings

Please add a reviewer when ready for review.

### CODE REVIEW

File: src/science/canopy/cbl_LAI_canopy_height.F90 [hunk 1]
```diff
@@ -1,26 +1,66 @@
...
+  ! This SUBROUTINE checks that input values of leaf area and canopy height
+  ! lie between set values (defined by PFT) and overwrites if necessary.
+  ! These limits are needed to ensure that the canopy roughness properties of
+  ! the land point are well behaved (see [[ruff_resist]]).
+  !
+  ! The mapping from grid cells (land_pts,ntiles) to the (mp) structure
+  ! is also undertaken.
+  !
+  ! **This SUBROUTINE is active run when CABLE is run within a coupled model
+  ! (i.e. ACCESS)**
```
Comment (ccarouge) [inline src/science/canopy/cbl_LAI_canopy_height.F90:22]:
```suggestion
  ! **This SUBROUTINE is active when CABLE is run within a coupled model
  ! (i.e. ACCESS)** 
```

File: src/science/canopy/cbl_LAI_canopy_height.F90 [hunk 2]
```diff
@@ -1,26 +1,66 @@
...
+  ! **This SUBROUTINE is active run when CABLE is run within a coupled model
+  ! (i.e. ACCESS)** 
+  !
+  !## Method
+  !
+  ! The cell-level input (land_pts,ntile) variables for leaf area `LAI_pft` and
+  ! canopy height `HGT_pft` are checked for
+  !
+  ! - whether the land_pt has a non-zero fraction of that tile (if zero
+  !   fraction then LAI and canopy height are set to zero.
```
Comment (ccarouge) [inline src/science/canopy/cbl_LAI_canopy_height.F90:30]:
```suggestion
  ! - whether the land_pt has a non-zero fraction of that tile (if zero
  !   fraction then LAI and canopy height are set to zero)
```

File: src/science/canopy/cbl_LAI_canopy_height.F90 [hunk 3]
```diff
@@ -1,26 +1,66 @@
...
+  ! - whether the LAI lies above a minimum value `CLAI_thresh`
+  ! - whether the canopy height lies a minimum value (which are PFT dependent)
```
Comment (ccarouge) [inline src/science/canopy/cbl_LAI_canopy_height.F90:32]:
```suggestion
  ! - whether the canopy height lies above a minimum value (which is PFT dependent)
```

File: src/science/canopy/cbl_LAI_canopy_height.F90 [hunk 4]
```diff
@@ -1,26 +1,66 @@
...
+  ! - whether non-vegetated tiles have a non-zero canopy height.
+  !
+  ! Outputs are `LAI_pft_cbl` and `HGT_pft_cbl`
+  !
+  ! **WARNINGS**
+  !
+  ! - INTENT statements need to be added to the argument lists
+  ! - hardwired indexing is use throughout.  This SUBROUTINE assumes that
```
Comment (ccarouge) [inline src/science/canopy/cbl_LAI_canopy_height.F90:40]:
```suggestion
  ! - hardwired indexing is used throughout.  This SUBROUTINE assumes that
```

File: src/science/canopy/cbl_LAI_canopy_height.F90 [hunk 5]
```diff
@@ -1,26 +1,66 @@
...
-integer :: tile_pts(ntiles) 
-integer:: tile_index(land_pts,ntiles) 
-real:: tile_frac(land_pts,ntiles) 
-logical :: L_tile_pts(land_pts,ntiles) 
-integer :: mp
-real :: LAI_pft(land_pts, ntiles)
-real :: HGT_pft(land_pts, ntiles)
+real :: Clai_thresh                 !! The minimum LAI below which a tile is considered to be NOT vegetated (m\(^2\)m\(^{-2}\))
+integer :: land_pts, ntiles         !! number of land points, max number of tiles that can be carried in any land point (-)
+integer :: tile_pts(ntiles)         !! number of land_pts that a particular tile occupies (-)
```
Comment (ccarouge) [inline src/science/canopy/cbl_LAI_canopy_height.F90:52]: 1 tile can cover several land points??? I don't think I understand here.

File: src/science/canopy/cbl_LAI_canopy_height.F90 [hunk 6]
```diff
@@ -1,26 +1,66 @@
...
+integer :: mp                       !! total number of tiles carried by land vector (-)
+real :: LAI_pft(land_pts, ntiles)   !! IN leaf area (m\(^2\)m\(^{-2}\))
+real :: HGT_pft(land_pts, ntiles)   !! IN canopy height (m)
 
 !return vars
-real :: HGT_pft_temp(land_pts,ntiles)
-real :: LAI_pft_cbl(mp)
-real :: HGT_pft_cbl(mp)
+real :: HGT_pft_temp(land_pts,ntiles) !! OUT adjusted canopy height in (land_pts,ntile) form (m)
+real :: LAI_pft_cbl(mp)             !! OUT adjusted leaf area in (mp) form (m^\(^2\)m\(^{-2}\))
```
Comment (ccarouge) [inline src/science/canopy/cbl_LAI_canopy_height.F90:62]:
```suggestion
real :: LAI_pft_cbl(mp)             !! OUT adjusted leaf area in (mp) form (m\(^2\)m\(^{-2}\))
```

**Review by ccarouge — CHANGES_REQUESTED** [review]
A few minor edits to go through

File: src/science/canopy/cbl_LAI_canopy_height.F90 (hunk 5 above)
Comment (har917) [reply src/science/canopy/cbl_LAI_canopy_height.F90:52]: okay - this is a bit confused.  I THINK that 

- each land_pt has up to max number of tiles as subunits (ntiles)
- but ntiles < number of tile types considered (=17 currently)

tile_pts(ntiles) is then number of land_pts which include a tile with a particular value of the tile type and tile_index is the mapping between (land_pts,ntile) and the tile type.

I will attempt to revise that comment accordingly

**Review by ccarouge — APPROVED** [review]

**Pull request merged**